framework/scripts/agent_upgrade.py

```diff
@@ -90,17 +89,36 @@ def get_agents_versions(agents: list) -> dict:
     dict
         Dictionary with the current version (prev_version).
     """
-    agents_versions = dict()
-    for agent_id in agents:
-        agent = Agent(agent_id)
-        agent.load_info_from_db()
-        if agent.version:
-            agents_versions[agent_id] = {
-                'prev_version': agent.version,
-                'new_version': None
-            }
+    f_kwargs = {
+        "agent_list": agents,
+        "select": ["version"],
+        "limit": len(agents)
+    }
+    agent_versions = await cluster_utils.forward_function(get_agents, f_kwargs=f_kwargs)
+    return {agent['id']: {"prev_version": agent['version'], "new_version": None}
+            for agent in agent_versions.affected_items}
 
-    return agents_versions
+
+async def get_agent_version(agent_id: str) -> str:
+    """Get the given agent's current version.
+
+    Parameters
+    ----------
+    agent_id : str
+        Agent ID.
+
+    Returns
+    -------
+    str
+        Agent version.
+    """
+    f_kwargs = {
+        "agent_list": [agent_id],
+        "select": ["version"],
+        "limit": 1
```

Comment:
Same as above

Reply:
Memory efficiency as the response will be lighter.